File: blinkreceipt-demo/BlinkReceiptStaticScan/src/main/java/com/blinkreceipt/ocr/BlinkApplication.java
package com.blinkreceipt.ocr;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;

import com.microblink.BlinkReceiptSdk;
import com.microblink.core.InitializeCallback;

public class BlinkApplication extends Application {

    private static final String TAG = "BlinkReceiptStaticScan";

    @Override
    public void onCreate() {
        super.onCreate();

        BlinkReceiptSdk.initialize(this, new InitializeCallback() {
            @Override
            public void onComplete() {
                Log.d(TAG, "BlinkReceiptSdk initialized!!!");
            }

            @Override
            public void onException(@NonNull Throwable throwable) {
                Log.e(TAG, "failure in initialize", throwable);
            }
        });
    }

    @Override
    public void onTerminate() {
        BlinkReceiptSdk.terminate();

        super.onTerminate();
    }

}

File: blinkreceipt-demo/BlinkReceiptBarcode/src/main/java/com/blinkreceipt/barcode/BlinkApplication.kt
package com.blinkreceipt.barcode

import android.app.Application
import android.util.Log
import com.microblink.barcode.BlinkReceiptBarcodeSdk
import com.microblink.core.InitializeCallback

class BlinkApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        BlinkReceiptBarcodeSdk.initialize(this, object : InitializeCallback {

            override fun onComplete() {
                Log.d(TAG, "initialize complete" )
            }

            override fun onException(e: Throwable) {
                Log.e(TAG, "failed in initialize", e )
            }

        })
    }

    private companion object {
        const val TAG = "BarcodeApplication"
    }
}